lib/fastlane_core/configuration/config_item.rb

```diff
@@ -63,16 +66,37 @@ def valid?(value)
 
     # Returns an updated value type (if necessary)
     def auto_convert_value(value)
-      # Special treatment if the user specififed true, false or YES, NO
-      if %w(YES yes true).include?(value)
-        value = true
-      elsif %w(NO no false).include?(value)
-        value = false
+      # We must cast the type to String before comparing because of the way that Ruby
+      # handles class comparisons
+      case data_type.to_s
+      when 'Array'
+        value = value.split(',')
```

Comment:
Good point. I was making the assumption that the command line option parser would not be automatically converting to integers, but in any case, I agree that we should be forcing type conversion at this point.